### PR TITLE
Removing card requirement for OC

### PR DESCRIPTION
**Pull request by wjrosa**
Fixes STRIPE-646

## Changes proposed in this Pull Request:

In this PR I am removing the requirement for the credit card payment method when the Optimized Checkout is enabled. This way, merchants are allowed to use the feature with any payment method setup they want, without having to force CCs.

## Testing instructions

- Checkout to this branch on your test environment (`fix/removing-card-requirement-for-oc`)
- Connect your Stripe account
- Enable the Optimized Checkout
- Disable the credit card payment method
- Enable any other payment method(s)
- As a shopper, add a product to your cart
- Go to the checkout page
- Confirm the OC container is rendered, showing the enabled payment methods inside of it
- Confirm you are able to complete the purchase

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [ ] Tested on mobile (or does not apply)

### Changelog entry

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

File: client/settings/general-settings-section/payment-method.js [hunk 1]
```diff
@@ -138,7 +137,7 @@ const PaymentMethod = ( { method, data } ) => {
 			// eslint-disable-next-line camelcase
 			wc_stripe_settings_params.has_klarna_gateway_plugin );
 
-	const isDisabledButChecked = PAYMENT_METHOD_CARD === method && isOCEnabled;
+	const isDisabledButChecked = false;
```
Comment (annemirasol) [inline client/settings/general-settings-section/payment-method.js]: Should we just remove this variable, and the `disabledButChecked` prop below it? It looks like it no longer has any use case for it.

Comment (wjrosa) [reply client/settings/general-settings-section/payment-method.js]: Sure! Removed it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4573/commits/d87a91939bcb20c338920d76d279a7e6148599ce

File: client/settings/general-settings-section/payment-method-description.js [hunk 2]
```diff
@@ -5,7 +5,6 @@ import RecurringPaymentIcon from '../../components/recurring-payment-icon';
 import PaymentMethodCapabilityStatusPill from 'wcstripe/components/payment-method-capability-status-pill';
 import PaymentMethodDeprecationPill from 'wcstripe/components/payment-method-deprecation-pill';
 import PaymentMethodUnavailableDueConflictPill from 'wcstripe/components/payment-method-unavailable-due-conflict-pill';
-import PaymentMethodRequiredForOCPill from 'wcstripe/components/payment-method-required-for-oc-pill';
```
Comment (malithsen) [inline client/settings/general-settings-section/payment-method-description.js:8]: We should also remove the component and the test now that this is no longer used.

**Review by malithsen — COMMENTED** [review]
Edge case: If the merchant disables all payment methods, we show the following error to the shopper

Image

I think we should hide the OC container if all payment methods are disabled.

File: client/settings/general-settings-section/payment-method-description.js (hunk 2 above)
Comment (wjrosa) [reply client/settings/general-settings-section/payment-method-description.js:8]: I was going to remove it in another PR (https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4584) but just merged it here already 👍

**Comment by wjrosa** [comment]
> Edge case: If the merchant disables all payment methods, we show the following error to the shopper
> 
> Image
> I think we should hide the OC container if all payment methods are disabled.

Good catch, @malithsen ! Hiding it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4573/commits/467aef1d7fe58a2682eabde5e260d394f0622e05

**Comment by annemirasol** [comment]
> > Edge case: If the merchant disables all payment methods, we show the following error to the shopper
> > Image
> > I think we should hide the OC container if all payment methods are disabled.
> 
> Good catch, @malithsen ! Hiding it in [467aef1](https://github.com/woocommerce/woocommerce-gateway-stripe/commit/467aef1d7fe58a2682eabde5e260d394f0622e05)

This is what I see when there are no enabled payment methods:

Screenshot 2025-08-14 at 10 07 28 AM

**Comment by wjrosa** [comment]
> > > Edge case: If the merchant disables all payment methods, we show the following error to the shopper
> > > Image
> > > I think we should hide the OC container if all payment methods are disabled.
> > 
> > 
> > Good catch, @malithsen ! Hiding it in [467aef1](https://github.com/woocommerce/woocommerce-gateway-stripe/commit/467aef1d7fe58a2682eabde5e260d394f0622e05)
> 
> This is what I see when there are no enabled payment methods:
> 
> Screenshot 2025-08-14 at 10 07 28 AM

I saw a merge issue after your latest merge to develop. Fixed it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4573/commits/44973f1b3c665604ec5efc22ea380ad0b0b7a714. Any chance that was your issue? I cannot reproduce the error message in your screenshot. This is what I see:

#### Block checkout

Screenshot 2025-08-14 at 15 56 51

#### Shortcode checkout

Screenshot 2025-08-14 at 15 56 40

**Review by malithsen — COMMENTED** [review]
I get the "no payment methods available" error when all pms are disabled (as expected). Nice work!

However, Link is not presented on this branch, and it seems like the e2e failures are related to this.

**Comment by wjrosa** [comment]
Thanks, Malith! I have fixed the issue in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4573/commits/4cdbfacec35c101639de61f620fb49d54639da6e. As we spoke on Slack, e2e are passing now for the main suite, but still failing here for the OC ones 👀 (they work locally)

**Review by malithsen — APPROVED** [review]
This looks good to me now! OC test timeouts don't seem related. I tested the flows manually in my local environment and it's working as expected. We should still have a look at those failures and address them in a separate PR.

**Review by annemirasol — APPROVED** [review]
Non-blocking, and possibly a separate issue, but there's a disconnect between what wp-admin and checkout says:

Screenshot 2025-08-19 at 8 57 25 AM

**Comment by wjrosa** [comment]
Thanks, Anne! I have created STRIPE-678 to handle this specific issue, since it may require some additional investigation.
